src/components/TodoHeader.tsx

```diff
@@ -0,0 +1,88 @@
+import { useEffect, useRef } from 'react';
+import { Todo } from '../types/Todo';
+
+type Props = {
+  setNewTodo: (value: string) => void;
+  newTodo: string;
+  setCurrentTodos: (value: React.SetStateAction<Todo[]>) => void;
+  setUpdatingTodo: (value: boolean) => void;
+  setErrorNotification: (value: string) => void;
+  updatingTodo: boolean;
+};
+
+export const Header: React.FC<Props> = ({
+  newTodo,
+  setNewTodo,
+  setCurrentTodos,
+  setUpdatingTodo,
+  setErrorNotification,
+  updatingTodo,
+}) => {
+  const inputRef = useRef<HTMLInputElement | null>(null);
+
+  useEffect(() => {
+    if (inputRef.current && !updatingTodo) {
+      inputRef.current.focus();
+    }
+  }, [updatingTodo]);
+
+  const handleInputChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+    setNewTodo(e.target.value);
+  };
+
+  const handleSubmit = (e: React.FormEvent) => {
+    e.preventDefault();
+    if (newTodo.trim().length > 0) {
+      setUpdatingTodo(true);
+      setTimeout(() => {
+        setCurrentTodos((prevTodos: Todo[]) => [
+          ...prevTodos,
+          {
+            id: new Date().getTime(),
+            title: newTodo,
+            completed: false,
+            userId: 0,
+          },
+        ]);
+        setNewTodo('');
+        setUpdatingTodo(false);
+      }, 300);
```

Comment:
I cannot see where you implement this
<img width="413" alt="image" src="https://github.com/mate-academy/react_todo-app-add-and-delete/assets/49764881/f5b73224-ad8e-4e59-9ff3-2b58c1831aa1">